src/math/Euler.js

```diff
@@ -243,6 +243,101 @@ class Euler {
 
 	}
 
+	// Based on: Bernardes E, Viollet S (2022) Quaternion to Euler angles conversion:
+	// A direct, general and computationally efficient method. PLoS ONE 17(11): e0276302.
+	setFromQuaternionDirect( q, order, update ) {
+
+        switch ( order ) {
+
+            case 'XYZ':
+
+                this._y = Mathutils.acos( clamp( -2 * q.w * q.y - 2 * q.x * q.z, -1, 1 ) );
```

Comment:
It's `Math.acos()`.

And switch to tabs.

This may help you: http://zz85.github.io/mrdoobapproves/

Reply:
Pretty cool website! Just changed that.